app/web_framework/cluster_master_application.py

```diff
@@ -206,10 +207,14 @@ def initialize(self, route_node=None, cluster_master=None):
         super().initialize(path=None)  # we will not set the root path until the get() method is called
 
     def get(self, build_id, path=None):
-        artifact_file_path = self._cluster_master.get_path_for_build_results_archive(int(build_id))
-        self.root, artifact_filename = os.path.split(artifact_file_path)
-        self.set_header('Content-Type', 'application/octet-stream')  # this should be downloaded as a binary file
-        return super().get(path=artifact_filename)
+        try:
+            artifact_file_path = self._cluster_master.get_path_for_build_results_archive(int(build_id))
+            self.root, artifact_filename = os.path.split(artifact_file_path)
+            self.set_header('Content-Type', 'application/octet-stream')  # this should be downloaded as a binary file
+            return super().get(path=artifact_filename)
+        except ItemNotReadyError as ex:
+            self.set_status(status_code=http.client.ACCEPTED, reason=str(ex))
+            return
```

Comment:
Actually, I just noticed that `ItemNotReadyError` is already in that dict! It looks like that's not working as expected. Ah, it's because `_BuildResultHandler` does not inherit from `ClusterBaseHandler`.

Hmm, this is probably fine then -- although I wonder if we should add a comment explaining why the exception needs to be caught here. (I think I've been confused by this before. :wink:)